lint-staged.config.js

```diff
@@ -0,0 +1,3 @@
+module.exports = {
+  "*.{ts,tsx}": [() => "npm run validate:precommit"],
```

Comment:
We should probably include JS, just to be safe. Let me know if you have a reason not to

```suggestion
  "*.{js,jsx,ts,tsx}": [() => "npm run validate:precommit"],
```